app/src/main/java/com/saucelabs/mydemoapp/android/view/fragments/LoginFragment.java

```diff
@@ -179,86 +174,42 @@ private String getMockBiometricUserName() {
 
     @SuppressLint("NewApi")
     private void loginWithEmail() {
-        if (binding.nameET.getText().toString().trim().isEmpty()) {
+        String username = binding.nameET.getText().toString().trim();
+        String password = binding.passwordET.getText().toString().trim();
+
+        if (username.isEmpty()) {
             binding.nameRL.setBackground(getActivity().getDrawable(R.drawable.edit_bg_red));
             binding.nameErrorTV.setVisibility(View.VISIBLE);
             binding.usernameErrorIV.setVisibility(View.VISIBLE);
             binding.passwordRL.setBackground(getActivity().getDrawable(R.drawable.edit_bg_grey));
             binding.passwordErrorTV.setVisibility(View.INVISIBLE);
             binding.passwordErrorIV.setVisibility(View.INVISIBLE);
-        } else if (binding.passwordET.getText().toString().trim().isEmpty()) {
+        } else if (password.isEmpty()) {
             binding.nameRL.setBackground(getActivity().getDrawable(R.drawable.edit_bg_grey));
             binding.nameErrorTV.setVisibility(View.INVISIBLE);
             binding.usernameErrorIV.setVisibility(View.INVISIBLE);
             binding.passwordRL.setBackground(getActivity().getDrawable(R.drawable.edit_bg_red));
             binding.passwordErrorTV.setVisibility(View.VISIBLE);
             binding.passwordErrorIV.setVisibility(View.VISIBLE);
             binding.passwordErrorTV.setText(getString(R.string.enter_password));
-        } else if (binding.nameET.getText().toString().contains("alice@example.com")) {
+        } else if (username.contains("alice@example.com")) {
             binding.passwordErrorTV.setText(getString(R.string.soory_this_user_has_been_locked_out));
             binding.passwordErrorTV.setVisibility(View.VISIBLE);
 
-            fetch404();
-
             TestFairy.addEvent("Authentication failed for locked user");
         } else {
-            fetchWiki();
-
             TestFairy.setUserId(binding.nameET.getText().toString().trim());
             TestFairy.addEvent("User signed in with password");
 
-//            ST.isLogin = true;
+            ST.isLogin = true;
+            ST.hasVisualChanges = username.equals("visual@example.com");
+
             if (mParam1.equals(ST.CHECKOUT)) {
                 Bundle bundle = ST.getBundle(MainActivity.FRAGMENT_CHECKOUT_INFO, 1);
                 ST.startActivityWithDataBundle(mAct, MainActivity.class, bundle, ST.START_ACTIVITY_WITH_FINISH);
-            } else
+            } else {
                 ST.startActivity(mAct, MainActivity.class, ST.START_ACTIVITY_WITH_CLEAR_BACK_STACK);
+            }
         }
     }
-
```

Comment:
Please restore these, these are documented in how to test network capture with RDC, and will be required again with System-wide Network Capture.

Reply:
Done